serving/docker/lmi.Dockerfile

```diff
@@ -85,78 +60,46 @@ COPY scripts scripts/
+    && mkdir -p /opt/ml/model \
+    && mkdir -p /opt/djl/bin
 COPY config.properties /opt/djl/conf/config.properties
 COPY partition /opt/djl/partition
+COPY scripts/telemetry.sh /opt/djl/bin
+RUN echo "${djl_serving_version} lmi" > /opt/djl/bin/telemetry
 
 COPY distribution[s]/ ./
 RUN mv *.deb djl-serving_all.deb || true
 
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -yq libaio-dev libopenmpi-dev g++ \
-    && scripts/install_openssh.sh \
-    && scripts/install_djl_serving.sh $djl_version $djl_serving_version \
-    && scripts/install_djl_serving.sh $djl_version $djl_serving_version ${djl_torch_version} \
-    && djl-serving -i ai.djl.onnxruntime:onnxruntime-engine:$djl_version \
-    && djl-serving -i com.microsoft.onnxruntime:onnxruntime_gpu:$onnx_version \
-    && scripts/install_python.sh ${python_version} \
-    && scripts/install_s5cmd.sh x64 \
-    && mkdir -p /opt/djl/bin && cp scripts/telemetry.sh /opt/djl/bin \
-    && echo "${djl_serving_version} lmi" > /opt/djl/bin/telemetry \
-    && pip3 cache purge \
-    && apt-get clean -y && rm -rf /var/lib/apt/lists/*
-
-RUN pip3 install torch==${torch_version} torchvision==${torch_vision_version} --index-url https://download.pytorch.org/whl/cu124
-RUN pip3 install \
-    ${seq_scheduler_wheel} \
-    peft==${peft_version} \
-    protobuf==${protobuf_version} \
-    transformers==${transformers_version} \
-    hf-transfer \
-    zstandard \
-    datasets==${datasets_version} \
-    mpi4py \
-    sentencepiece \
-    tiktoken \
-    blobfile \
-    einops \
-    accelerate==${accelerate_version} \
-    bitsandbytes==${bitsandbytes_version} \
-    auto-gptq==${auto_gptq_version} \
-    pandas \
-    pyarrow \
-    jinja2 \
-    retrying \
-    opencv-contrib-python-headless \
-    safetensors \
-    scipy \
-    onnx \
-    sentence_transformers \
-    onnxruntime \
-    autoawq==${autoawq_version} \
-    tokenizers==${tokenizers_version} \
-    pydantic==${pydantic_version} \
-    ${djl_converter_wheel} \
-    optimum==${optimum_version} \
-    ${flash_infer_wheel} \
-    ${vllm_wheel} \
-    ${lmi_dist_wheel} \
-    torch==${torch_version} \
-    torchvision==${torch_vision_version} \
-    ${sagemaker_fast_model_loader_wheel} \
-    && git clone https://github.com/neuralmagic/AutoFP8.git && cd AutoFP8 && git reset --hard 4b2092c && pip3 install . && cd .. && rm -rf AutoFP8 \
-    && pip3 cache purge
-
+RUN apt-get update && apt-get install -yq libaio-dev libopenmpi-dev g++ unzip
+RUN scripts/install_openssh.sh
+RUN scripts/install_python.sh ${python_version}
+RUN scripts/install_s5cmd.sh x64
+RUN pip3 cache purge
 # Add CUDA-Compat
 RUN apt-get update && apt-get install -y cuda-compat-12-4 && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+RUN apt-get clean -y && rm -rf /var/lib/apt/lists/*
+
+COPY requirements-lmi.txt ./requirements.txt
+RUN pip3 install torch==2.5.1 torchvision==0.20.1 --index-url https://download.pytorch.org/whl/cu124
```

Comment:
Agreed on reducing number of layers - 1 and 3 are somewhat at odds and we're trying to balance these. 

For caching/pulling purposes, we want to ensure layers at the bottom (top of the docker file) remain constant, and put the more mutable layers at the bottom. We also sometimes need to separate commands into multiple layers here to facilitate this.

In general I've tried to address this in https://github.com/deepjavalibrary/djl-serving/pull/2591. The biggest contributor to our image size (aside from base image) our the python dependencies (of which pytorch is 5GB out of 10GB). I think it makes sense to put pytorch early and in its own layer so that it can be cached effectively and not redownloaded when we do djl-serving updates or security updates.